### PR TITLE
Material navigation

### PR DESCRIPTION
**Pull request by marcelinkaaa**
@gubatron Just wanted you to take a look. Still a long way to go, but hopefully its a good start. Issues:
- white screen on app start - I always seen a delay in app launched and we had a blue background with the action bar with logo to fill it in. Not sure if I introduced that white screen while the app is loading. Wanted to create a splash/welcome screen with our blue background and full logo in the center but it did not work out just now. 
- we still need the initial search fragment to be loaded on start 
- the green update ready icon will be invisible when there is no update, and when clicked it should show the update dialog
- 'My music' is not fully connected to the app, crashes upon switching
- Upon trying to download a file the app crashes. Not properly connected to transfers. 
- 'exit' shows the dialog and exits but when a user clicks on 'cancel' on the dialog we see a blank fragment with title exit instead of previously loaded fragment

Basically I did not understand quite a bit of code in MainActivity and I know many things need to be adjusted to work with new navigation, I just get stuck on what to do. 

Could you let me know if I'm on the right path, and help out?

screenshot_20170216-143913

### CODE REVIEW

**Comment by gubatron** [comment]
I'll take a look, would love this merged soon because the app right now has no mention of either "FrostWire" nor the logo is to be seen once you open it.

**Comment by gubatron** [comment]
So the first thing is that you got rid of the SearchFragment in the layout, but you didn't put it back in the XML anywhere else, might want to think where you want that to be put.

File: android/src/com/frostwire/android/gui/activities/MainActivity.java [hunk 1]
```diff
@@ -204,6 +285,48 @@ public void run() {
...
+    }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_SEARCH) {
+            if (!(getCurrentFragment() instanceof SearchFragment)) {
+                controller.switchFragment(R.id.menu_main_search);
+            }
+        } else if (keyCode == KeyEvent.KEYCODE_MENU) {
+//            toggleDrawer();
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/MainActivity.java]: hmm, gotta have something like this again.
Although I haven't seen any new androids with this key, users that do have it will feel the pain of not being able to pull out the menu with the menu key.

**Comment by gubatron** [comment]
I'm afraid this patch will be just illustratory @marcelinkaaa there's way too many changes, specially in MainActivity that I don't feel comfortable with. Next time I see you, let's work together on replacing things. I feel this will break too much.

**Comment by marcelinkaaa** [comment]
@gubatron of course. I just wanted you to go over the layout and all the changes needed to be made to insert new drawer. If you ok with all of them - I can create a new branch to include all the new code for the nav drawer without any bigger changes to MainActivity so you can take over that part and make sure all is working properly.
I needed to make those changes so I can preview the nav drawer and style it correctly.

File: android/src/com/frostwire/android/gui/activities/MainActivity.java [hunk 2]
```diff
@@ -139,54 +141,133 @@
...
+        setContentView(R.layout.activity_main);
         this.controller = new MainController(this);
-        this.fragmentsStack = new Stack<>();
-        this.permissionsCheckers = initPermissionsCheckers();
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
+        setSupportActionBar(toolbar);
+
+        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/MainActivity.java]: might want to keep this and use it on the `toggleDrawer()` method to be invoked on `KeyEvent.KEYCODE_MENU`

File: android/src/com/frostwire/android/gui/activities/MainActivity.java [hunk 3]
```diff
@@ -139,54 +141,133 @@
...
+                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        drawer.addDrawerListener(toggle);
+        toggle.syncState();
+
+        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+        View navHeaderView =  navigationView.getHeaderView(0);
+        TextView title = (TextView)navHeaderView.findViewById(R.id.nav_header_main_title);
+        String basicOrPlus = Constants.IS_GOOGLE_PLAY_DISTRIBUTION ? "Basic" : "Plus";
+        title.setText("FrostWire " + basicOrPlus + " v" + Constants.FROSTWIRE_VERSION_STRING);
+        TextView buildNumber = (TextView)navHeaderView.findViewById(R.id.nav_header_main_build);
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/MainActivity.java]: 👍

File: android/src/com/frostwire/android/gui/activities/MainActivity.java [hunk 4]
```diff
@@ -708,64 +759,35 @@ private void refreshMenuRemoveAdsItem() {
...
-            restoreFragmentsStack(savedInstanceState);
-        }
-        if (fragment == null) {
-            fragment = search;
-            setCheckedItem(R.id.menu_main_search);
-        }
-
-        switchContent(fragment);
-    }
+//    private void setupInitialFragment(Bundle savedInstanceState) {
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/MainActivity.java]: hmm, commenting this out might be causing issues.

File: android/src/com/frostwire/android/gui/activities/MainActivity.java [hunk 5]
```diff
@@ -918,48 +886,11 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
...
-                activity.syncSlideMenu();
-            }
-        }
-    }
 
     public void performYTSearch(String ytUrl) {
         SearchFragment searchFragment = (SearchFragment) getFragmentByMenuId(R.id.menu_main_search);
         searchFragment.performYTSearch(ytUrl);
-        switchContent(searchFragment);
+//        switchContent(searchFragment);
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/MainActivity.java]: :)

File: android/src/com/frostwire/android/gui/activities/internal/MainController.java [hunk 6]
```diff
@@ -45,9 +45,9 @@ public MainActivity getActivity() {
         return activity;
     }
 
-    public void closeSlideMenu() {
-        activity.closeSlideMenu();
-    }
+//    public void closeSlideMenu() {
```
Comment (gubatron) [inline android/src/com/frostwire/android/gui/activities/internal/MainController.java]: not necessary anymore? does it close all by itself in all situations?

**Comment by gubatron** [comment]
there's a lot we'll reuse, perhaps it can be fixed, but it will require a ton of testing to be sure we can use this branch.

**Comment by aldenml** [comment]
@marcelinkaaa thanks for pushing this, it's wild :) but give us a good hint of what to use for the "new drawer"

**Comment by marcelinkaaa** [comment]
@aldenml will be working from Angel's tomorrow, so wanted to go over the best way of implementing with existing code. I know its a mess in the activity :)

**Comment by gubatron** [comment]
Opening new one from a 2-commit branch.

One commit to make the replacement of the menu.
Another commit to refactor main activity into decoupling the menu into a `NavigationMenu` class.
